src/main/java/com/synopsys/integration/alert/workflow/upgrade/step/_4_0_0/conversion/CommonDistributionFieldCreator.java

```diff
@@ -0,0 +1,81 @@
+import com.synopsys.integration.alert.common.enumeration.FrequencyType;
+import com.synopsys.integration.alert.database.api.configuration.model.ConfigurationFieldModel;
+import com.synopsys.integration.alert.database.deprecated.channel.CommonDistributionConfigEntity;
+import com.synopsys.integration.alert.database.deprecated.channel.CommonDistributionRepository;
+import com.synopsys.integration.alert.database.entity.repository.NotificationTypeRepository;
+import com.synopsys.integration.alert.database.relation.repository.DistributionNotificationTypeRepository;
+
+@Component
+public class CommonDistributionFieldCreator {
+    private final CommonDistributionRepository commonDistributionRepository;
+    private final DistributionNotificationTypeRepository distributionNotificationTypeRepository;
+    private final NotificationTypeRepository notificationTypeRepository;
+    private final FieldCreatorUtil fieldCreatorUtil;
+
+    @Autowired
+    public CommonDistributionFieldCreator(final CommonDistributionRepository commonDistributionRepository, final DistributionNotificationTypeRepository distributionNotificationTypeRepository,
+        final NotificationTypeRepository notificationTypeRepository, final FieldCreatorUtil fieldCreatorUtil) {
+        this.commonDistributionRepository = commonDistributionRepository;
+        this.distributionNotificationTypeRepository = distributionNotificationTypeRepository;
+        this.notificationTypeRepository = notificationTypeRepository;
+        this.fieldCreatorUtil = fieldCreatorUtil;
+    }
+
+    public List<ConfigurationFieldModel> createCommonFields(final String descriptorName, final Long configId) {
+        final CommonDistributionConfigEntity commonDistributionEntity = commonDistributionRepository.findByDistributionConfigIdAndDistributionType(configId, descriptorName);
+        final List<ConfigurationFieldModel> configurationFieldModels = new LinkedList<>();
+
+        final String name = commonDistributionEntity.getName();
+        final String storedDescriptorName = commonDistributionEntity.getDistributionType();
+        final FrequencyType frequency = commonDistributionEntity.getFrequency();
+        final String providerName = commonDistributionEntity.getProviderName();
+
+        fieldCreatorUtil.addFieldModel(CommonDistributionUIConfig.KEY_NAME, name, configurationFieldModels);
+        fieldCreatorUtil.addFieldModel(CommonDistributionUIConfig.KEY_CHANNEL_NAME, storedDescriptorName, configurationFieldModels);
+        fieldCreatorUtil.addFieldModel(CommonDistributionUIConfig.KEY_FREQUENCY, frequency, configurationFieldModels);
+        fieldCreatorUtil.addFieldModel(CommonDistributionUIConfig.KEY_PROVIDER_NAME, providerName, configurationFieldModels);
+
+        return configurationFieldModels;
+    }
+
+    /* FIXME We need to also take out the provider specific info from CommonDistributionConfigEntity and store it into the BD provider distribution configuration
```

Comment:
Shouldn'y this be done in this PR?

Reply:
We will want to do this once our grouped configuration stuff has been fully implemented. I'm working a new PR right now to do all that stuff